### PR TITLE
Add function to irregularly rebin data in N dimensions (Part 2) -Correct branch

### PR DESCRIPTION
**Pull request by DanRyanIrish**

### CODE REVIEW

File: stixpy/product/tools.py [hunk 1]
```diff
@@ -156,7 +164,28 @@ def rebin_irregular(cube, axes_idx_edges, operation=np.mean, operation_ignores_m
...
-                new_coord_values[k] = np.stack(tmp_coords[k], axis=l)
+                idx_i = np.where(coord_arr_idx == i)[0][0]
+                new_coord_values[k] = np.stack(tmp_coords[k], axis=idx_i)
+
+    # Build new WCS.
+    ec = ExtraCoords()
+    for name, axis, coord, physical_type in zip(new_coord_names, new_coord_pix_idxs,
+                                                new_coord_values, new_coord_physical_types):
+        if len(axis) != 1:
+            raise NotImplementedError("Rebinning multi-dimensional coordinates not supported.")
```
Comment (samaloney) [inline stixpy/product/tools.py:175]: Nice so this is really the only limitation of this method 😎

**Review by samaloney — APPROVED** [review]
Nice looks good to me, ~~I assume the docs fail is because not merge with current main?~~ It because of the updates to xrayvision I think need to merge #115.

File: stixpy/product/tests/test_tools.py [hunk 2]
```diff
@@ -28,24 +28,30 @@ def cube():
...
-                                                              0.14142135623730953,
-                                                              0.14142135623730953]])
-    expected_coords = [[5.] * u.s, [3., 4.5, 6.5] * u.keV]
+    data = np.array([[1., 2., 2.]])
+    mask = np.zeros(data.shape, dtype=bool)
+    uncertainty = astropy.nddata.StdDevUncertainty([[0.1,
+                                                     0.14142135623730953,
+                                                     0.14142135623730953]])
+    ec = ExtraCoords()
+    ec.add('iso(utc; None', 1, [5.] * u.s, physical_types="time")
```
[the remark this answers is not in the document]
Comment (DanRyanIrish) [reply stixpy/product/tests/test_tools.py:44]: As ugly as this look, it's correct :)

**Comment by DanRyanIrish** [comment]
I pulled upstream/main so that's why the diff has exploded.

**Pull request merged**